sklearn/preprocessing/_function_transformer.py

```diff
@@ -77,6 +80,22 @@ def __init__(self, func=None, inverse_func=None, validate=True,
         self.kw_args = kw_args
         self.inv_kw_args = inv_kw_args
 
+    def _check_functions(self, X):
+        """Check that transform and inverse_transform lead to a 2D array"""
+        idx_selected = slice(None, None, max(1, X.shape[0] // 100))
```

Comment:
Why not just select a couple of samples? 1% seems an extraordinarily large amount to test the output shape!